CMakeLists.txt

```
@@ -127,7 +127,12 @@ else()
 endif()
 
 if(MSVC AND NOT DEFINED CMAKE_MSVC_RUNTIME_LIBRARY)
-    set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+    if(NOT DEFINED BUILD_SHARED_LIBS OR NOT BUILD_SHARED_LIBS)
```

```suggestion
    if(NOT BUILD_SHARED_LIBS)
```
If `BUILD_SHARED_LIBS` is undefined it will evaluate to false anyway.

---

In any case,[ Conan already seems to define `CMAKE_MSVC_RUNTIME_LIBRARY`](https://github.com/conan-io/conan/blob/59029095e86eacd402083fca7c308b06889f6301/conan/tools/cmake/toolchain/blocks.py#L56-L114), why is the extra condition here necessary?

---

This is for building with plain cmake or from vcpkg, conan handles this for us.

---

[vcpkg also sets it](https://github.com/microsoft/vcpkg/blob/055721089e8037d4d617250814d11f881e557549/scripts/toolchains/windows.cmake#L3)

I guess the intention is to default to the shared runtime library instead of the static one? The shared one is default behavior, so I think you could simplify this to `if (not defined CMAKE_MSVC_RUNTIME_LIBRARY) set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL") endif()`, but we can also do this in a second pass just to clarify our behavior around `CMAKE_MSVC_RUNTIME_LIBRARY`.